### PR TITLE
Upgrade to prompt tool kit 2.0

### PR DESCRIPTION
**Pull request by MrMeemus**
Upgrade to prompt tool kit 2.0 with initial smoke/sanity testing.

### CODE REVIEW

**Comment by MrMeemus** [comment]
Referencing these awesome PRs :) :
https://github.com/dbcli/pgcli/pull/930 
https://github.com/dbcli/pgcli/pull/826
https://github.com/dbcli/mycli/pull/666

File: mssqlcli/mssql_cli.py [hunk 1]
```diff
@@ -279,28 +280,23 @@ def handle_editor_command(self, cli, document):
...
+            query = (special.get_editor_query(text) or
                      self.get_last_query())
             sql, message = special.open_external_editor(filename, sql=query)
             if message:
                 # Something went wrong. Raise an exception and bail.
                 raise RuntimeError(message)
-            cli.current_buffer.document = Document(sql, cursor_position=len(sql))
-            cli.application.pre_run_callables = []
-            document = cli.run()
+            text = self.prompt_session.prompt(default=sql)
```
Comment (meeuw) [inline mssqlcli/mssql_cli.py]: This requires a loop now (like in pgcli) or else ctrl-c will abort mssql-cli.

**Comment by j-bennet** [comment]
In the toolbar, options that are ON are not highlighted:

image

They are in master:

image

**Comment by MrMeemus** [comment]
Merging this change with known bug of key bindings not being highlighted. Will open a new issue to track this.

**Pull request merged**